packages/ui/src/components/va-data-table/hooks/useSortable.ts

```diff
@@ -129,6 +130,7 @@ export const useSortable = (
   // Sets the clicked heading's column as a one to sort by and toggles the sorting order from "asc" to "desc" to `null`
   // (un-sorted) if the same column is clicked again or sets sorting order to "asc" if some other column is chosen.
   function toggleSorting (column: DataTableColumnInternal) {
+    sortedColumn.value = column.key
```

Comment:
I'd make another event and call it "columnSorted". @update:sortingOrder should be used only for updating sortingOrder prop.